lib/broccoli/ember-addon.js

```diff
@@ -19,8 +19,20 @@ module.exports = EmberAddon;
   @constructor
   @param options
 */
-function EmberAddon(options) {
-  options = options || {};
+function EmberAddon() {
+  var args = [];
+  var options = {};
+
+  for (var i = 0, l = arguments.length; i < l; i++) {
+    args.push(arguments[i]);
+  }
+
+  if (args.length === 1) {
+    options = args[0];
+  } else if (args.length > 1) {
+    options = defaults.apply(null, args);
```

Comment:
We use a slightly different pattern in ember-app for deeply merged values ([see here](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L199-L212)), which is something like:

```javascript
var defaults      = require('lodash/object/defaults');
var merge         = require('lodash/object/merge');

var defaults      = require('lodash/object/defaults');
var merge         = require('lodash/object/merge');

options = merge({}, args, defaults);
```